### ISSUE TITLE
Terraform TLS Handshake Timeout errors

### ISSUE THREAD

**Issue by mwarkentin** [Homebrew/homebrew-core#8576]
# Please follow the general troubleshooting steps first:

- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] Confirmed this is problem with specific formulae and not Homebrew/brew? If it's a general Homebrew/brew problem please file this issue at https://github.com/Homebrew/brew/issues/new

### Bug reports:

Summary: Terraform 0.8.2 installed via homebrew gets TLS handshake timeout errors, terraform 0.8.2 downloaded from terraform.io works correctly. Several people on our team saw the same issue with the brew installed version only.

```
Error refreshing state: 8 error(s) occurred:
* cloudflare_record.primary: Error finding zone "*****.com": ListZones command failed: error from makeRequest: HTTP request failed: Get https://api.cloudflare.com/client/v4/zones?name=******.com: net/http: TLS handshake timeout
...
```

Perhaps it has something to do with the version of Go used to build it or some other dependency?

* https://github.com/hashicorp/terraform/issues/10718#issuecomment-266946488
* https://github.com/hashicorp/terraform/issues/11007

### Brew Config

```
$ brew config
HOMEBREW_VERSION: 1.1.6-23-g1a394aa
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: 1a394aa7e5b7bf4f6281826248345be0308b30a4
Last commit: 14 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 74063d85f4da579d02e34d8770756273707aec0b
Core tap last commit: 8 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: quad-core 64-bit broadwell
Homebrew Ruby: 2.0.0-p648
Clang: 8.0 build 800
Git: 2.8.1 => /usr/local/bin/git
Perl: /usr/bin/perl
Python: /usr/local/var/pyenv/shims/python => /usr/local/var/pyenv/versions/2.7.11/bin/python2.7
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
Java: 1.8.0_102
macOS: 10.11.6-x86_64
Xcode: 8.1
CLT: 8.2.0.0.1.1480973914
X11: N/A
```

### Brew Doctor

Note: brew doctor is reporting my xcode is out of date, however I don't see any upgrades available. This might be because I'm still on El Cap instead of Sierra or something?

![](https://mwarkentin-snaps.s3.amazonaws.com/Monosnap_2017-01-06_18-47-47.png)

```
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry and just ignore them. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and what additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew provided
script of the same name. We found the following "config" scripts:
  /usr/local/var/pyenv/shims/python-config
  /usr/local/var/pyenv/shims/python2-config
  /usr/local/var/pyenv/shims/python2.7-config
  /usr/local/var/pyenv/shims/python3-config
  /usr/local/var/pyenv/shims/python3.5-config
  /usr/local/var/pyenv/shims/python3.5m-config

Warning: Your Xcode (8.1) is outdated.
Please update to Xcode 8.2 (or delete it).
Xcode can be updated from the App Store.
```

**Comment by mwarkentin** [comment]
CC @mitchellh @stack72

Sorry not really sure of a better way of getting eyes on this. :)

**Comment by mwarkentin** [comment]
@MikeMcQuaid Will try to figure out the xcode thing, thanks!

**Comment by mwarkentin** [comment]
@MikeMcQuaid I'm not familiar with how the "bottled" versions of homebrew items are created - is it possible to see what version of Go it would have been built with? Maybe it just needs a "re-bottle" now with Go 1.7.4 if that's not what it was built with originally?

**Comment by woodruffw** [comment]
Terraform 0.8.2 was bottled on December 22, and Go 1.7.4 was bottled on December 1 (a54add9af12cbd2676a21e7af0029c12fadd11f9), so the current Terraform should be built on it.

**Comment by woodruffw** [comment]
@#

**Comment by mwarkentin** [comment]
@woodruffw Thanks, good to be able to remove that as a possibility.

**Comment by ilovezfs** [comment]
Could be because the resource blocks haven't been updated since November 1st?

**Comment by mitchellh** [comment]
I want to chime in and note that we confirmed this to be an issue caused by Go 1.7.4. The fix is to use Go 1.7.3 or earlier. Unfortunate, but that is the status of things currently.

**Comment by ilovezfs** [comment]
@mitchellh thanks for the update. Yeah, that is not good news since we only have go 1.7.4.

**Comment by mwarkentin** [comment]
Some related links:

* https://github.com/golang/go/issues/18141
* https://github.com/golang/go/issues/18203
* https://go-review.googlesource.com/#/c/34389/

**Comment by ilovezfs** [comment]
Closed via #8628 and #8654.

**Comment by mwarkentin** [comment]
Thanks for the quick fix on this!
